compiler/src/dotty/tools/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -238,11 +238,16 @@ trait BCodeSkelBuilder extends BCodeHelpers {
 
       val ps = claszSymbol.info.parents
       val superClass: String = if (ps.isEmpty) ObjectReference.internalName else internalName(ps.head.widenDealias.typeSymbol)
-      val interfaceNames = classBTypeFromSymbol(claszSymbol).info.interfaces map {
+      val interfaceNames0 = classBTypeFromSymbol(claszSymbol).info.interfaces map {
         case classBType =>
           if (classBType.isNestedClass) { innerClassBufferASM += classBType }
           classBType.internalName
       }
+      val interfaceNames =
+        if (claszSymbol.is(ModuleClass) && !interfaceNames0.contains("java/io/Serializable"))
```

Comment:
We should inline the commit message from https://github.com/lampepfl/dotty/commit/33b989478c1b0ee8ce69691f9dde427ba94d8dab here to explain what's going on:
```suggestion
        // To avoid deadlocks when combining objects, lambdas and multi-threading,
        // lambdas in objects are compiled to instance methods of the module class
        // instead of static methods (see tests/run/deadlock.scala and
        // https://github.com/scala/scala-dev/issues/195 for details).
        // This has worked well for us so far but this is problematic for
        // serialization: serializing a lambda requires serializing all the values
        // it captures, if this lambda is in an object, this means serializing the
        // enclosing object, which fails if the object does not extend
        // Serializable.
        // Because serializing objects is basically free since #5775, it seems like
        // the simplest solution is to simply make all objects Serializable, this
        // certainly seems preferable to deadlocks.
        if (claszSymbol.is(ModuleClass) && !interfaceNames0.contains("java/io/Serializable"))
```
And maybe add an extra note about the fact that this can't be done earlier because scala.js won't like it.